test/functional/services/test_subjects.ts

```diff
@@ -199,6 +199,58 @@ export function TestSubjectsProvider({ getService }: FtrProviderContext) {
       });
     }
 
+    public async setValueWithChecks(
+      selector: string,
+      text: string,
+      options: SetValueOptions = {}
+    ): Promise<void> {
+      return await retry.try(async () => {
+        const { clearWithKeyboard = false, typeCharByChar = false } = options;
+        log.debug(`TestSubjects.setValueWithChecks(${selector}, ${text})`);
+        await this.click(selector);
+        // in case the input element is actually a child of the testSubject, we
+        // call clearValue() and type() on the element that is focused after
+        // clicking on the testSubject
+        const input = await find.activeElement();
+
+        await retry.tryForTime(5000, async () => {
+          let currentValue = await input.getAttribute('value');
+          if (currentValue !== '') {
+            if (clearWithKeyboard === true) {
+              await input.clearValueWithKeyboard();
+            } else {
+              await input.clearValue();
+            }
+            currentValue = await input.getAttribute('value');
+          }
+
+          if (currentValue === '') {
+            return true;
+          } else {
+            throw new Error(`Expected input to be empty, but got value '${currentValue}'`);
+          }
+        });
+
+        for (const chr of text) {
+          await retry.tryForTime(5000, async () => {
+            const oldValue = await input.getAttribute('value');
+            await input.type(chr, { charByChar: typeCharByChar });
+
+            await retry.tryForTime(1000, async () => {
+              const newValue = await input.getAttribute('value');
+              if (newValue === `${oldValue}${chr}`) {
+                return true;
+              } else {
+                throw new Error(
+                  `After typing character '${chr}', the new value in the input should be '${oldValue}${chr}' (got ${newValue})`
+                );
+              }
+            });
+          });
+        }
```

Comment:
I'm not sure if this solution will be effective and stable:
- if the problem is that the letter may not be typed, it makes sense to reimplement  `WebElementWrapper: type function` for ML tests. Wrapping existing code with 2 retries will slow things down with check will not give a benefit if the wrong letter or in the wrong place will be typed. 
- if the problem is that the field may be cleared during typing I suggest checking how the component re-rendering is triggered. Wrapping things with double retry will most like hide an issue the real user can face.

Reply:
But 500 passing tests means the opposite :)

Reply:
@dmlemeshko thanks for the feedback!
Some explanations to this approach:
- We're already investigating the component and will (hopefully) fix it as soon as we know why it behaves that way, so this PR is only a temporary solution to fix typing until then (this is the 1st reason why I've implemented this as a separate method).
- The slowdown due to retries is expected. I've called the method `setValueWithChecks` to set the right expectations: when I cal something **WithChecks**, I expect it to be slower (2nd reason for a separate method).
- I agree that the outer retry is probably not needed (but I think the same holds true for the `setValue` method). This will just retry the whole thing in case an error is thrown on the way, so it doesn't really hurt. For the inner retries:
  - The first block will retry to clear the field
  - The second block does basically the following:
    a. Type in the next character
    b. Check if it was appended to the input value (give it 1 second time to appear there)
    c. If it doesn't appear, try again, starting with step a. (retry for 5 seconds, then fail)
  - So I agree that it doesn't help to fix wrong letters or wrong order of letters. But that was not the scope of this fix, it should only check for missing letters.
- I can see how this new method `setValueWithChecks` is very specialized to (hopefully temporarily) fix a specific issue and might not be appropriate for the `testSubjects` service. Happy to move it to a ML specific location if you think that's better.

Reply:
@pheyos thank you for explaining the issue, it makes more sense now. Since you mentioned it is a temporary fix for ML-specific components, I suggest we don't put it in `testSubjects.ts`: we probably don't want other teams to start using it?

Reply:
@dmlemeshko agreed. I've moved the method to a common ML service in a11a340 .